### PR TITLE
parallel and stashing tutorials

### PR DESCRIPTION
**Pull request by zhreshold**
*Issue #, if available:*

*Description of changes:*

Parallel jenkins build_docs stage

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-787-11 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-787/11/index.html

File: Jenkinsfile [hunk 1]
```diff
@@ -275,6 +275,198 @@ stage("Unit Test") {
   }
 }
 
+stage("Build Tutorials") {
```
Comment (szha) [inline Jenkinsfile:278]: looking at this file, I think you'd start benefiting from refactoring the pipelines into functions.

**Comment by szha** [comment]
Job PR-787-12 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-787/12/index.html

File: Jenkinsfile (hunk 1 above)
Comment (zhreshold) [reply Jenkinsfile:247]: any working example? I tried global functions but hit something similar to https://github.com/jenkins-docs/creating-a-pipeline-in-blue-ocean/issues/19

**Comment by szha** [comment]
Job PR-787-16 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-787/16/index.html

**Comment by zhreshold** [comment]
The parallel tutorial build is working.
However, per @szha suggestion, it can be improved for readability if the shared functions can be used. 
Please advice if we should wait for solutions or proceed.

**Review by Innixma — APPROVED** [review]
CI end-to-end in under an hour?! Count me in!

I think we can do a cleanup to avoid code duplication as a follow-up PR, which I would recommend we do soon so it isn't forgotten.

Another thing I see is that the Text module is taking ~40 minutes for its doc build, @sxjscience do you think you could reduce the tutorial runtime to ~10 minutes as a follow-up PR? This would allow our end-to-end CI to finish in just **_30 minutes_**, which would be awesome.

**Comment by sxjscience** [comment]
Yes, we may change to use less trials.

**Pull request merged**